schemas/gen.py

```diff
@@ -134,19 +134,30 @@ def gen_go_schemas_package(schemas: List[Schema]) -> List[str]:
     return lines
 
 
-def next_struct_name(file: str, start: int) -> str:
+def next_type_name(file: str, start: int) -> str:
     """
     Find the name of the next struct definition in a go file starting at a given line.
 
+    It is pretty dumb and only matches type definitions of the form:
+
+        ^type Thing struct.*   // a struct definition
+
+        ^type Thing.*          // a non-struct definition
+
     This is how we decide which struct to operate on for the //go:generate comments above structs.
     """
     with open(file) as f:
         for lineno, line in enumerate(f.readlines()):
             if lineno < start:
                 continue
+            # Detect struct types.
             match = re.match("type ([\\S]+) struct", line)
             if match is not None:
                 return match[1]
+            # Detect non-struct types.
+            match = re.match("type ([\\S]+)", line)
+            if match is not None:
+                return match[1]
```

Comment:
wouldn't 157-160 match everything 153-156 matches, plus non-structs? can 153-156 be removed?